functions/src/utils.ts

```diff
@@ -129,19 +136,26 @@ export function isValidUrl(urlString: string | undefined): boolean {
 }
 
 /**
- * Verify a Firebase ID token is valid.
+ * Verify a Firebase ID token is valid and belongs to an @actnowcoalition.org account.
  *
  * @param token Firebase ID token to verify.
  * @returns True if the token is valid, throws an error otherwise.
  */
-export async function verifyIdToken(token: string | undefined) {
+ export async function verifyIdToken(token: string | undefined) {
   if (!token) {
     throw new ShareLinkError(ShareLinkErrorCode.INVALID_TOKEN);
   }
   return firebaseApp
     .auth()
     .verifyIdToken(token)
-    .then(() => true)
+    .then((token) => {
```

Comment:
Checking that the token belongs to an `@actnowcoalition.org` address here means that we should be able to keep our Firestore rules strictly as `allow read, write: if false` and let the serverside SDK handle all interactions with storage